File: samples/email-a-poem/src/main/java/io/quarkiverse/langchain4j/sample/EmailMeAPoemResource.java
package io.quarkiverse.langchain4j.sample;

import jakarta.ws.rs.GET;
import jakarta.ws.rs.Path;

@Path("/email-me-a-poem")
public class EmailMeAPoemResource {

    private final MyAiService service;

    public EmailMeAPoemResource(MyAiService service) {
        this.service = service;
    }

    @GET
    public String emailMeAPoem() {
        return service.writeAPoem("Quarkus", 4);
    }

}

File: samples/email-a-poem-kotlin/src/main/kotlin/io/quarkiverse/langchain4j/sample/EmailMeAPoemResource.kt
package io.quarkiverse.langchain4j.sample

import jakarta.ws.rs.GET
import jakarta.ws.rs.Path

@Path("/email-me-a-poem")
class EmailMeAPoemResource(private val service: MyAiService) {
    @GET
    fun emailMeAPoem(): String {
        return service.writeAPoem("Quarkus", 4)
    }
}
